RedHat-UX/red-hat-design-system
# docs: component status table refactor

Closes #973

## What I did

1. Updated the CSV file to reflect the new table structure
2. Updated the `componentStatus` shortcode template according to the design spec
3. Added `csv` as a new data extension for eleventy to detect and parse
4. Since we were no longer parsing the CSV at the time we were adding the elementDocs collection, I needed to pull the componentStatus global data into it via the `collectionApi` (referenced: https://github.com/11ty/eleventy/issues/775#issuecomment-671322087).
5. I didn't see why the componentStatus shortcode had to be a paired shortcode, so I removed that and updated all instances. It just seemed cleaner to me.

## Testing Instructions

1. Verify that I migrated the CSV file data correctly. WebRH maps to Drupal/FTS repo and WebDMS maps to studio repo.
2. Verify that the tables appear in all the spots where the shortcode is declared and that it matches the design spec in the linked issue.

## Notes to Reviewers

1. I was tempted to rename everything from componentStatus to repoStatus to simplify the transition, but let me know what you think too.
